clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -646,7 +646,14 @@ LogicalResult LowerFunction::buildFunctionEpilog(const LowerFunctionInfo &FI) {
         if (auto al = findAlloca(ret)) {
           rewriter.replaceAllUsesWith(al.getResult(), RVAddr);
           rewriter.eraseOp(al);
+          rewriter.setInsertionPoint(ret);
           rewriter.replaceOpWithNewOp<ReturnOp>(ret);
+
+          auto retInputs = ret.getInput();
```

Comment:
You are replacing `ret` and then looking at its input, I'm surprised this doesn't crash, please retrieve anything belonging to `ret` before you replace it.

Reply:
good catch!